### PR TITLE
frieren: wider FFN (mlp-ratio=8) for richer per-block representation

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The SOTA backbone uses `--model-mlp-ratio 4` (default), giving each Transformer block a FFN of width 4×512=2048. The MLP is the primary "memory" of each block — it stores per-token transformations independently of the attention bottleneck. A wider FFN at ratio=8 (4096 neurons per block) should allow each block to fit more complex non-linear transformations from the slice-based attention aggregates, potentially improving surface pressure and wall-shear fidelity without adding any new architectural complexity.

This is a single, low-risk knob: the model parameter count grows ~50% (roughly an extra 10M params, from ~20M to ~30M) but forward/backward memory impact is modest since the widening is in the FFN, not in the attention (no additional key/value caches). The cost is small and the potential gain is significant — deeper FFN expressivity is one of the least-explored levers remaining in the current architecture.

**Prediction:** mlp_ratio=8 should improve all axes roughly uniformly (better representation capacity) with the largest gains on tau_y/tau_z (currently 3.4× above AB-UPT), since those asymmetric shear components likely require higher-order geometry correlations.

## Instructions

Apply the following **single change** to the SOTA command. Everything else is identical to the SOTA baseline (PR #115):

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent frieren \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --model-mlp-ratio 8 \
  --ema-decay 0.999 \
  --wandb-group tay-round13-mlp-ratio-8
```

The **only** change from SOTA is `--model-mlp-ratio 8` (was the default 4). All other flags are locked.

Note: `--volume-loss-weight` is intentionally omitted here (same as the SOTA PR #115 run — the BASELINE note says PR #115 did NOT use `vol_w=2.0`; this keeps it comparable).

## Expected behaviour

- Training memory: roughly constant (FFN weights are not kept in GPU memory during forward by default in PyTorch — only activations matter for memory, and GELU activations are ≈ rank-1 in the FFN dimension so peak memory increase is modest).
- Throughput: expect ~15–20% slowdown per epoch (more ops in FFN layers). With budget of 270 min you should complete 7–8 epochs. Focus on `val/abupt_axis_mean_rel_l2_pct` trajectory.
- If the run diverges before epoch 3 or val_abupt > 20 at epoch 3, kill it and report.

## Baseline to beat

Current SOTA from PR #115 (thorfinn), W&B run `d03oghpp`:

| Metric | SOTA (PR #115) | AB-UPT |
|---|---:|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **9.484** | — |
| `test_primary/abupt_axis_mean_rel_l2_pct` | **10.580** | — |
| `test_primary/surface_pressure_rel_l2_pct` | **5.690** | 3.82 |
| `test_primary/wall_shear_rel_l2_pct` | **10.419** | 7.29 |
| `test_primary/volume_pressure_rel_l2_pct` | **12.740** | 6.08 |
| `test_primary/wall_shear_x_rel_l2_pct` | **8.908** | 5.35 |
| `test_primary/wall_shear_y_rel_l2_pct` | **12.491** | 3.65 |
| `test_primary/wall_shear_z_rel_l2_pct` | **13.071** | 3.63 |

**Beat `val_primary/abupt_axis_mean_rel_l2_pct < 9.484` to be a winner.**

## Results

Post your epoch-by-epoch `val/abupt_axis_mean_rel_l2_pct` trajectory and the final full test metrics table in a comment on this PR.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Frieren — this PR has been open since assignment with no acknowledgment or W&B run ID posted. Please confirm you have picked this up.

Quick notes on the updated context:
- **New merge bar: val_abupt < 9.291%** (not 9.484% as stated in the PR body — PR #222 fern lr_warmup=1ep merged and raised the bar)
- The SOTA baseline command now includes `--lr-warmup-epochs 1` — add this to your run command:

```bash
torchrun --standalone --nproc_per_node=8 train.py \
  --agent frieren \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --model-mlp-ratio 8 \
  --ema-decay 0.999 \
  --lr-warmup-epochs 1 \
  --wandb-group tay-round13-mlp-ratio-8
```

Post your W&B run URL and epoch-1 val_abupt within 30 minutes of pickup. If ep3 val_abupt > 20%, kill and report. Kill threshold: ep5 > 13% kill. To beat baseline you need val_abupt < 9.291% at best epoch.

**Comment by morganmcg1** [comment]
ADVISOR: frieren — second status request. Your pod is still 1/1 ready but no W&B run ID has been posted on this PR. Either:

1. **Post your W&B run ID(s) and ep1 val_abupt** if you've launched, OR
2. **Describe what's blocking you** (merge conflict, env error, code issue) so we can unblock.

Reminder: SOTA baseline command now includes `--lr-warmup-epochs 1`. Use the updated baseline command from the PR body and apply only the `--model-mlp-ratio 8` delta.

If no response in the next hour, this assignment will be reassigned.

**Comment by morganmcg1** [comment]
ADVISOR: W&B scan confirms you ARE running — your pods are live and training, you just never posted run IDs to this PR.

**Discovered W&B runs (group: tay-round13-mlp-ratio-8):**
- rank0: `kobvnazq`
- 8 ranks total running

**Current training progress (mlp_ratio=8, tay-round13-mlp-ratio-8):**

| Epoch | val_abupt (frieren) | val_abupt (SOTA ep-for-ep) |
|---|---:|---:|
| ep1 | 54.62% | 67.73% |
| ep2 | 23.74% | 41.93% |
| ep3 | 15.99% | 19.30% |
| ep4 | **12.92%** | **13.73%** |

**🟢 FRIEREN IS TRACKING AHEAD OF SOTA at ep4 (+0.81pp lead).** This is a strong positive signal — mlp_ratio=8 may actually help.

Merge bar: val_abupt < 9.291%. Continue to ep9 and post final results here.

Please post your results comment when training completes.

**Comment by morganmcg1** [comment]
ADVISOR: Trajectory update — ep5 data in.

**Current state:** ep5 = 11.301% (step 13604), run still active.

**Convergence projection** (geometric decay, ratio 0.527 from recent epochs):

| Epoch | Projected val_abupt |
|---|---|
| ep5 (actual) | 11.301% |
| ep6 | 10.447% |
| ep7 | 9.997% |
| ep8 | 9.759% |
| ep9 | 9.634% |
| ep10 | 9.568% |
| ep11 | 9.533% |
| ep12 | 9.515% |

**Assessment:** Frieren has the most favorable decay ratio of the three slow-converging runs (0.527), giving it the best trajectory. However, even with optimistic extrapolation the asymptote appears around 9.5%, which is +0.44pp above SOTA (9.065%). The model would need to significantly accelerate its improvement rate in later epochs to catch the merge bar.

**mlp_ratio=8 effect:** The doubled MLP ratio (4→8) is adding capacity but not improving the convergence floor. More parameters per layer means more data needed to fully train — the timeout constraint is hurting this hypothesis more than others.

**Merge bar:** 9.065% (PR #232). Current projection cannot reach this bar.

**Verdict:** Likely negative result. There is a small chance of deviation from the geometric decay model in later epochs (some models see a secondary improvement phase). Please continue and report every epoch. If ep7 comes in at or below 9.8%, that would be a significant positive surprise worth watching.

**Comment by morganmcg1** [comment]
ADVISOR: Progress check at ~262min runtime. W&B shows ep5.8 val=9.772% — this is the strongest active run. Only 0.707pp above SOTA 9.065%. Final 1-2 epochs are your best shot at beating baseline. Trajectory is still descending. Post your final W&B run ID and best val epoch when training completes.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #240 because ADVISOR: NEGATIVE — does not beat SOTA.

Run kobvnazq (frieren/mlp-ratio-8) finished at ep7.52, runtime 270.9min (timeout-bounded).

Final val_primary/abupt_axis_mean_rel_l2_pct trajectory:
| Epoch | Step  | val %    | Gap to SOTA (9.065%) |
|-------|-------|----------|----------------------|
| 1.00  | 2720  | 54.6200  | +45.555 pp |
| 2.00  | 5441  | 23.7401  | +14.675 pp |
| 3.00  | 8162  | 15.9900  | +6.925 pp |
| 4.00  | 10883 | 12.9196  | +3.855 pp |
| 5.00  | 13604 | 11.3006  | +2.236 pp |
| 6.00  | 16325 | 10.4737  | +1.409 pp |
| 7.00  | 19046 | 9.7725   | +0.708 pp |
| **7.52** | **20444** | **9.5498** | **+0.485 pp** |

Verdict: Increasing the transformer mlp_ratio from 4 to 8 doubles MLP block capacity, and the trajectory was monotonically improving — but deceleration set in around ep5 (per-epoch delta dropped from -0.83pp to -0.22pp), so the run asymptoted near 9.5% within the 270min wall-clock cap. The capacity is real but the data/optimization budget required to amortize it exceeds the available timeout. Close as NEGATIVE; do not pursue further mlp_ratio scaling at this epoch budget..
